cli/crates/common/src/analytics.rs

```diff
@@ -0,0 +1,174 @@
+#![allow(clippy::let_underscore_untyped)] // derivative
+
+use crate::{environment::Environment, errors::CommonError};
+use chrono::{DateTime, Utc};
+use core::panic;
+use derivative::Derivative;
+use once_cell::sync::OnceCell;
+use rudderanalytics::{
+    client::RudderAnalytics,
+    message::{Message, Track},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::{path::PathBuf, thread};
+use ulid::Ulid;
+
+#[derive(Derivative, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[derivative(Debug)]
+pub struct Analytics {
+    #[derivative(Debug = "ignore")]
+    #[serde(skip)]
+    client: RudderAnalytics,
+    session_id: Ulid,
+    start_time: DateTime<Utc>,
+    version: String,
+}
+
+// TODO move this to [`Environment`]
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[allow(clippy::module_name_repetitions)]
+pub struct GrafbaseConfig {
+    pub enable_analytics: bool,
+    pub anonymous_id: Option<Ulid>,
+}
+
+impl ToString for GrafbaseConfig {
```

Comment:
You should implement `std::fmt::Display` instead, which then gives you automatically `ToString`.

Reply:
I'm never sure of this since `Display` feels like a debug / visualization trait while `ToString` is more serialization

Reply:
Display is to display something. It allows you to use this in every formatting string, and gives you additionally `ToString`. It's always preferrable over `ToString`.

Reply:
Wouldn't `Display` be more of a debugging trait? e.g. the difference between outputting minified JSON and running Prettier, human-centric output